src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -178,6 +178,21 @@ void Terminal::UpdateSettings(winrt::Microsoft::Terminal::Settings::ICoreSetting
         proposedTop -= (proposedBottom - bufferSize.Y);
     }
 
+    Cursor& cursor = _buffer->GetCursor();
+    auto cursorPosition = cursor.GetPosition();
+    if (cursorPosition.X > viewportSize.X)
+    {
+        cursorPosition.X = viewportSize.X - 1;

```

Comment:
I'm confused as to why the X dimension is reset to Size.X-1 and the Y direction is reset to Size.Y exactly.

I would think that the rectangle is either inclusive in both dimensions, or exclusive on both directions.

Additionally, the std::min/std::max functions are great for situations like this as a short hand way of writing this.

Reply:
The runtime failure is caused by  https://github.com/microsoft/terminal/blob/429af0e6fa80412f1e9185845d856e3497a6cf78/src/buffer/out/textBufferCellIterator.cpp#L46

The `limits.IsInBounds` check is using `pos.X < RightExclusive()` so I have to use `Size.X - ` to pass it. Practially I've not seen crashes caused by `Size.Y`. But you are right, It should be unifed.